pkg/controller/core/workload_controller.go

```diff
@@ -159,6 +159,11 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		if evictionTriggered, err := r.reconcileCheckBasedEviction(ctx, &wl); evictionTriggered || err != nil {
 			return ctrl.Result{}, err
 		}
+
+		if reservationRemoved, err := r.reconcileReservationOnQueueDeletion(ctx, &wl, cqName); reservationRemoved || err != nil {
+			return ctrl.Result{}, err
+		}
```

Comment:
Uhm, I'm wondering if we should do this confirmation in L135. Because if the `reservationRemoved` is `true`, any other processes aren't needed.

WDYT?

Reply:
It was added here to have the "has reservations" cases grouped.

Reply:
why not before `reconcileCheckBasedEviction`?

Reply:
> It was added here to have the "has reservations" cases grouped.

This sounds fine.

This is just a comment, we need to wait for the next reconciliation triggered by workload's `conditions` updating if `SyncAdmittedCondition` returns true.